Error on import with latest released version

As part of the [JOSS review](https://github.com/openjournals/joss-reviews/issues/4904) I tried installing `autocti`. 

Though there were no errors during installation, I got an error in `welcome.py` which I can reproduce by simply doing `import autocti as ac`:

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mattcraig/mambaforge/envs/pyautoci/lib/python3.10/site-packages/autocti/__init__.py", line 10, in <module>
    from arcticpy import ROE
  File "/Users/mattcraig/mambaforge/envs/pyautoci/lib/python3.10/site-packages/arcticpy/__init__.py", line 1, in <module>
    from arcticpy.cti import add_cti, remove_cti, CTI_model_for_HST_ACS
  File "/Users/mattcraig/mambaforge/envs/pyautoci/lib/python3.10/site-packages/arcticpy/cti.py", line 8, in <module>
    from arcticpy.ccd import CCDPhase, CCD
  File "/Users/mattcraig/mambaforge/envs/pyautoci/lib/python3.10/site-packages/arcticpy/ccd.py", line 3, in <module>
    from autoconf.dictable import Dictable
ImportError: cannot import name 'Dictable' from 'autoconf.dictable' (/Users/mattcraig/mambaforge/envs/pyautoci/lib/python3.10/site-packages/autoconf/dictable.py)
```

## my installation steps

```bash
mamba create -n pyautoci python=3.10 scipy=1.8
conda activate pyautoci
pip install autocti
pip install arcticpy_no_gsl
```

@Jammy2211 

Looks like arcticpy also needs updating:

`ImportError while loading conftest '/home/runner/work/PyAutoCTI/PyAutoCTI/PyAutoCTI/test_autocti/conftest.py'.
test_autocti/conftest.py:6: in <module>
    import autocti as ac
autocti/__init__.py:10: in <module>
    from arcticpy import ROE
/opt/hostedtoolcache/Python/3.9.[18](https://github.com/Jammy2211/PyAutoCTI/actions/runs/6529920316/job/17728316475#step:10:19)/x64/lib/python3.9/site-packages/arcticpy/__init__.py:1: in <module>
    from arcticpy.cti import add_cti, remove_cti, CTI_model_for_HST_ACS
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/arcticpy/cti.py:8: in <module>
    from arcticpy.ccd import CCDPhase, CCD
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/arcticpy/ccd.py:3: in <module>
    from autoconf.dictable import Dictable
E   ImportError: cannot import name 'Dictable' from 'autoconf.dictable' (/home/runner/work/PyAutoCTI/PyAutoCTI/PyAutoConf/autoconf/dictable.py)`

--- a/autocti/charge_injection/model/analysis.py
+++ b/autocti/charge_injection/model/analysis.py
@@ -6,6 +6,7 @@ from autoconf import conf
 
 import autoarray as aa
 import autofit as af
+from autoconf.dictable import output_to_json
 
 from autocti.charge_injection.imaging.imaging import ImagingCI
 from autocti.charge_injection.fit import FitImagingCI
@@ -239,9 +240,9 @@ class AnalysisImagingCI(af.Analysis):
             visualization,and the pickled objects used by the aggregator output by this function.
         """
 
-        self.clocker.output_to_json(file_path=paths._files_path / "clocker.json")
-        self.settings_cti.output_to_json(
-            file_path=paths._files_path / "settings_cti.json"
+        output_to_json(self.clocker, file_path=paths._files_path / "clocker.json")
+        output_to_json(
+            self.settings_cti, file_path=paths._files_path / "settings_cti.json"
         )
 
         if conf.instance["visualize"]["plots"]["combined_only"]:
@@ -256,7 +257,8 @@ class AnalysisImagingCI(af.Analysis):
             cosmic_ray_map_path=dataset_path / "cosmic_ray_map.fits",
             overwrite=True,
         )
-        self.dataset.layout.output_to_json(
+        output_to_json(
+            self.dataset.layout,
             file_path=dataset_path / "layout.json",
         )
 
@@ -279,7 +281,8 @@ class AnalysisImagingCI(af.Analysis):
                 overwrite=True,
             )
 
-            self.dataset.layout.output_to_json(
+            output_to_json(
+                self.dataset.layout,
                 file_path=dataset_path / "layout.json",
             )
 

--- a/autocti/clocker/abstract.py
+++ b/autocti/clocker/abstract.py
@@ -4,12 +4,10 @@ from arcticpy import CCD
 from arcticpy import CCDPhase
 from arcticpy import TrapInstantCapture
 
-from autoconf.dictable import Dictable
-
 from autocti import exc
 
 
-class AbstractClocker(Dictable):
+class AbstractClocker:
     def __init__(self, iterations: int = 1, verbosity: int = 0):
         """
         An abstract clocker, which wraps the c++ arctic CTI clocking algorithm in **PyAutoCTI**.

--- a/autocti/dataset_1d/model/analysis.py
+++ b/autocti/dataset_1d/model/analysis.py
@@ -3,6 +3,7 @@ from typing import List, Optional
 from autoconf import conf
 
 import autofit as af
+from autoconf.dictable import output_to_json
 
 from autocti.dataset_1d.dataset_1d.dataset_1d import Dataset1D
 from autocti.dataset_1d.fit import FitDataset1D
@@ -145,7 +146,8 @@ class AnalysisDataset1D(af.Analysis):
             pre_cti_data_path=dataset_path / "pre_cti_data.fits",
             overwrite=True,
         )
-        self.dataset.layout.output_to_json(
+        output_to_json(
+            self.dataset.layout,
             file_path=dataset_path / "layout.json",
         )
         self.dataset.mask.output_to_fits(
@@ -161,16 +163,17 @@ class AnalysisDataset1D(af.Analysis):
                 pre_cti_data_path=dataset_path / "pre_cti_data.fits",
                 overwrite=True,
             )
-            self.dataset_full.layout.output_to_json(
+            output_to_json(
+                self.dataset_full.layout,
                 file_path=dataset_path / "layout.json",
             )
             self.dataset_full.mask.output_to_fits(
                 file_path=dataset_path / "mask.fits", overwrite=True
             )
 
-        self.clocker.output_to_json(file_path=paths._files_path / "clocker.json")
-        self.settings_cti.output_to_json(
-            file_path=paths._files_path / "settings_cti.json"
+        output_to_json(self.clocker, file_path=paths._files_path / "clocker.json")
+        output_to_json(
+            self.settings_cti, file_path=paths._files_path / "settings_cti.json"
         )
 
     def in_ascending_fpr_order_from(self, quantity_list, fpr_value_list):

--- a/autocti/model/model_util.py
+++ b/autocti/model/model_util.py
@@ -3,8 +3,6 @@ from typing import Optional, List
 from arcticpy import CCDPhase
 from arcticpy import TrapInstantCapture
 
-from autoconf.dictable import Dictable
-
 
 class AbstractCTI:
     @property
@@ -16,7 +14,7 @@ class AbstractCTI:
         return sum([trap.delta_ellipticity for trap in self.trap_list])
 
 
-class CTI1D(AbstractCTI, Dictable):
+class CTI1D(AbstractCTI):
     def __init__(
         self,
         trap_list: Optional[List[TrapInstantCapture]] = None,
@@ -43,7 +41,7 @@ class CTI1D(AbstractCTI, Dictable):
         return self._trap_list
 
 
-class CTI2D(AbstractCTI, Dictable):
+class CTI2D(AbstractCTI):
     def __init__(
         self,
         parallel_trap_list: Optional[List[TrapInstantCapture]] = None,
@@ -89,18 +87,12 @@ class CTI2D(AbstractCTI, Dictable):
 
 
 def is_parallel_fit(model):
-    if model.parallel_ccd is not None and model.serial_ccd is None:
-        return True
-    return False
+    return model.parallel_ccd is not None and model.serial_ccd is None
 
 
 def is_serial_fit(model):
-    if model.parallel_ccd is None and model.serial_ccd is not None:
-        return True
-    return False
+    return model.parallel_ccd is None and model.serial_ccd is not None
 
 
 def is_parallel_and_serial_fit(model):
-    if model.parallel_ccd is not None and model.serial_ccd is not None:
-        return True
-    return False
+    return model.parallel_ccd is not None and model.serial_ccd is not None

--- a/autocti/model/settings.py
+++ b/autocti/model/settings.py
@@ -2,12 +2,11 @@ from typing import List, Optional, Tuple
 
 from arcticpy import TrapInstantCapture
 
-from autoconf.dictable import Dictable
 
 from autocti import exc
 
 
-class AbstractSettingsCTI(Dictable):
+class AbstractSettingsCTI:
     def check_total_density_within_range_of_traps(
         self, total_density_range: Tuple[float, float], traps: List[TrapInstantCapture]
     ):

--- a/test_autocti/clocker/test_dict.py
+++ b/test_autocti/clocker/test_dict.py
@@ -1,63 +1,74 @@
 import os
 
 import pytest
+
+from autoconf.dictable import output_to_json, from_dict, from_json
 from autocti import Clocker2D
 
 
 @pytest.fixture(name="clocker_dict")
 def make_clocker_dict():
     return {
-        "type": "autocti.clocker.two_d.Clocker2D",
-        "iterations": 5,
-        "parallel_roe": {
-            "type": "arcticpy.roe.ROE",
-            "dwell_times": {
-                "type": "numpy.ndarray",
-                "array": [1.0],
-                "dtype": "float64",
+        "class_path": "autocti.clocker.two_d.Clocker2D",
+        "type": "instance",
+        "arguments": {
+            "iterations": 5,
+            "parallel_roe": {
+                "class_path": "arcticpy.roe.ROE",
+                "type": "instance",
+                "arguments": {
+                    "dwell_times": {
+                        "type": "numpy.ndarray",
+                        "array": [1.0],
+                        "dtype": "float64",
+                    },
+                    "empty_traps_between_columns": True,
+                    "empty_traps_for_first_transfers": False,
+                    "force_release_away_from_readout": True,
+                    "use_integer_express_matrix": False,
+                },
             },
-            "empty_traps_between_columns": True,
-            "empty_traps_for_first_transfers": False,
-            "force_release_away_from_readout": True,
-            "use_integer_express_matrix": False,
-        },
-        "parallel_express": 0,
-        "parallel_window_offset": 0,
-        "parallel_window_start": 0,
-        "parallel_window_stop": -1,
-        "parallel_poisson_traps": False,
-        "parallel_fast_mode": False,
-        "serial_roe": {
-            "type": "arcticpy.roe.ROE",
-            "dwell_times": {
-                "type": "numpy.ndarray",
-                "array": [1.0],
-                "dtype": "float64",
+            "parallel_express": 0,
+            "parallel_window_offset": 0,
+            "parallel_window_start": 0,
+            "parallel_window_stop": -1,
+            "parallel_poisson_traps": False,
+            "parallel_fast_mode": False,
+            "serial_roe": {
+                "type": "instance",
+                "class_path": "arcticpy.roe.ROE",
+                "arguments": {
+                    "dwell_times": {
+                        "type": "numpy.ndarray",
+                        "array": [1.0],
+                        "dtype": "float64",
+                    },
+                    "empty_traps_between_columns": True,
+                    "empty_traps_for_first_transfers": False,
+                    "force_release_away_from_readout": True,
+                    "use_integer_express_matrix": False,
+                },
             },
-            "empty_traps_between_columns": True,
-            "empty_traps_for_first_transfers": False,
-            "force_release_away_from_readout": True,
-            "use_integer_express_matrix": False,
+            "serial_express": 0,
+            "serial_window_offset": 0,
+            "serial_window_start": 0,
+            "serial_window_stop": -1,
+            "serial_fast_mode": False,
+            "verbosity": 0,
+            "poisson_seed": -1,
         },
-        "serial_express": 0,
-        "serial_window_offset": 0,
-        "serial_window_start": 0,
-        "serial_window_stop": -1,
-        "serial_fast_mode": False,
-        "verbosity": 0,
-        "poisson_seed": -1,
     }
 
 
 def test_clocker_from_dict(clocker_dict):
-    assert isinstance(Clocker2D.from_dict(clocker_dict), Clocker2D)
+    assert isinstance(from_dict(clocker_dict), Clocker2D)
 
 
 def test_file():
     filename = "/tmp/temp.json"
-    Clocker2D().output_to_json(filename)
+    output_to_json(Clocker2D(), filename)
 
     try:
-        assert isinstance(Clocker2D.from_json(filename), Clocker2D)
+        assert isinstance(from_json(filename), Clocker2D)
     finally:
         os.remove(filename)

--- a/test_autocti/model/test_model_util.py
+++ b/test_autocti/model/test_model_util.py
@@ -2,6 +2,7 @@ import pytest
 from os import path
 
 import autocti as ac
+from autoconf.dictable import output_to_json, from_json
 
 
 def test__trap_list():
@@ -97,9 +98,9 @@ def test__dictable():
         serial_ccd=serial_ccd,
     )
 
-    cti.output_to_json(file_path=json_file)
+    output_to_json(cti, file_path=json_file)
 
-    cti_from_json = ac.CTI2D.from_json(file_path=json_file)
+    cti_from_json = from_json(file_path=json_file)
 
     assert cti_from_json.parallel_trap_list[0].density == 1.0
     assert cti_from_json.parallel_trap_list[1].density == 2.0

--- a/test_autocti/model/test_settings_dict.py
+++ b/test_autocti/model/test_settings_dict.py
@@ -3,26 +3,30 @@ import os
 import pytest
 
 import autocti as ac
+from autoconf.dictable import from_dict, output_to_json, from_json
 
 
 @pytest.fixture(name="settings_dict")
 def make_settings_dict():
     return {
-        "type": "autocti.model.settings.SettingsCTI2D",
-        "parallel_total_density_range": None,
-        "serial_total_density_range": None,
+        "class_path": "autocti.model.settings.SettingsCTI2D",
+        "type": "instance",
+        "arguments": {
+            "parallel_total_density_range": None,
+            "serial_total_density_range": None,
+        },
     }
 
 
 def test_settings_from_dict(settings_dict):
-    assert isinstance(ac.SettingsCTI2D.from_dict(settings_dict), ac.SettingsCTI2D)
+    assert isinstance(from_dict(settings_dict), ac.SettingsCTI2D)
 
 
 def test_file():
     filename = "/tmp/temp.json"
-    ac.SettingsCTI2D().output_to_json(filename)
+    output_to_json(ac.SettingsCTI2D(), filename)
 
     try:
-        assert isinstance(ac.SettingsCTI2D().from_json(filename), ac.SettingsCTI2D)
+        assert isinstance(from_json(filename), ac.SettingsCTI2D)
     finally:
         os.remove(filename)

